sdks/python/apache_beam/transforms/cy_combiners.py

```diff
@@ -359,49 +348,40 @@ def add_input(self, element):
     self.count += 1
     self.sum += element
     bucket_index = self.calculate_bucket_index(element)
-    size_of_bucket = len(self.buckets)
-    self.increment_bucket(bucket_index)
-    if size_of_bucket == 0:
-      self.first_bucket_offset = bucket_index
-    else:
-      self.first_bucket_offset = min(self.first_bucket_offset, bucket_index)
+    self.buckets[bucket_index] += 1
+    self.first_bucket_offset = min(self.first_bucket_offset, bucket_index)
+    self.last_bucket_offset = max(self.last_bucket_offset, bucket_index)
 
   def calculate_bucket_index(self, element):
-    """Calculate the bucket index for the given element"""
     if element == 0:
       return 0
     log10_floor = get_log10_round_to_floor(element)
-    power_of_ten = math.pow(10, log10_floor)
-    if element < 2 * power_of_ten:
-      bucket_offset = 0  # [0, 2)
-    elif element < 5 * power_of_ten:
-      bucket_offset = 1  # [2, 5)
+    power_of_ten = POWER_TEN[log10_floor]
+    if element < power_of_ten * 2:
+      bucket_offset = 0
+    elif element < power_of_ten * 5:
+      bucket_offset = 1
     else:
-      bucket_offset = 2  # [5, 10)
-    return 1 + (log10_floor * self.buckets_per_10) + bucket_offset
-
-  def increment_bucket(self, bucket_index):
-    """Increment the bucket for the given index
-    If the bucket at the given index is already in the list,
-    this will increment the existing value.
-    If the specified index is outside of the current bucket range,
-    the bucket list will be extended to incorporate the new bucket
-    """
-    if not self.buckets:
-      self.buckets.append(1)
-    elif bucket_index < self.first_bucket_offset:
-      new_buckets = []
-      new_buckets.append(1)
-      new_buckets.extend(
-          [0] * (self.first_bucket_offset - bucket_index - 1))
-      self.buckets = new_buckets + self.buckets
-    elif bucket_index >= self.first_bucket_offset + len(self.buckets):
-      self.buckets.extend(
-          [0] * (bucket_index - self.first_bucket_offset - len(self.buckets)))
-      self.buckets.append(1)
-    else:
-      self.buckets[bucket_index - self.first_bucket_offset] += 1
-
-
-class DistributionCounterFn(AccumulatorCombineFn):
-  _accumulator_type = DistributionAccumulator
+      bucket_offset = 2
+    return 1 + log10_floor * self.buckets_per_10 + bucket_offset
+
+  def translate_to_histogram(self, histogram):
```

Comment:
<!--new_thread; commit:dd3ae5364c173a99c8a15f7e80960b9307993f7c; resolved:0-->
Please add doc, in particular with what the type of `histogram` should be.

A side note is that this is a weird pattern, where we need to pass in a preallocated `histogram`.  Can you also document why we use this pattern instead of returning a newly created `Histogram`?

Reply:
Histogram is a class under ```runners/dataflow/internal/clients```.  According to what @robertwb suggested, it's better to avoid importing runners/dataflow into modules outside of the scope.